spec/models/filtered_question_spec.rb

```diff
@@ -15,66 +15,85 @@
+        expect(filtered_question(q2, d0)).not_to be_visible
+        expect(filtered_question(q2_a, d0)).not_to be_visible
+        expect(filtered_question(q2_b, d0)).not_to be_visible
+        expect(filtered_question(q21, d0)).not_to be_visible
+
+        # Middle-generation division selected
+        expect(filtered_question(q0, d1)).to be_visible
+        expect(filtered_question(q1, d1)).to be_visible
+        expect(filtered_question(q11, d1)).not_to be_visible
+        expect(filtered_question(q2, d1)).not_to be_visible
+        expect(filtered_question(q2_a, d1)).not_to be_visible
+        expect(filtered_question(q2_b, d1)).not_to be_visible
+        expect(filtered_question(q21, d1)).not_to be_visible
+
+        # Leaf division selected
+        expect(filtered_question(q0, d11)).to be_visible
+        expect(filtered_question(q1, d11)).to be_visible
+        expect(filtered_question(q11, d11)).to be_visible
+        expect(filtered_question(q2, d11)).not_to be_visible
+        expect(filtered_question(q2_a, d11)).not_to be_visible
+        expect(filtered_question(q2_b, d11)).not_to be_visible
+        expect(filtered_question(q21, d11)).not_to be_visible
+
+        # Middle-generation division selected
+        expect(filtered_question(q0, d2)).to be_visible
+        expect(filtered_question(q1, d2)).not_to be_visible
+        expect(filtered_question(q11, d2)).not_to be_visible
+        expect(filtered_question(q2, d2)).to be_visible
+        expect(filtered_question(q2_a, d2)).to be_visible
+        expect(filtered_question(q2_b, d2)).not_to be_visible
+        expect(filtered_question(q21, d2)).not_to be_visible
+
+        # Leaf division selected
+        expect(filtered_question(q0, d21)).to be_visible
+        expect(filtered_question(q1, d21)).not_to be_visible
+        expect(filtered_question(q11, d21)).not_to be_visible
+        expect(filtered_question(q2, d21)).to be_visible
+        expect(filtered_question(q2_a, d21)).to be_visible
+        expect(filtered_question(q2_b, d21)).to be_visible
+        expect(filtered_question(q21, d21)).to be_visible
+      end
+    end
+
+    context 'with top division user' do
+      let(:user) { create(:user, :admin, division: d0) }
+
+      it_behaves_like 'full visibility'
+    end
+
+    context 'with system user' do
+      let(:user) { :system }
+
+      it_behaves_like 'full visibility'
     end
 
     context 'with user in lower division' do
-      let!(:user) { create(:user, :admin, division: d1) }
+      let(:user) { create(:user, :admin, division: d1) }
 
       it 'should still be able to see questions from ancestor division' do
+        # This is usually not permitted for most objects. Loan questions are different.
         expect(filtered_question(q0, d0)).to be_visible
       end
+
+      it 'should not be able to see questions in divisions outside own acestors and descendants' do
+        expect(filtered_question(q2, d2)).not_to be_visible
```

Comment:
I think this is an illegal option—it's saying this user (who belongs to d1) selected d2 in the division select menu. They shouldn't be able to see d2 in that menu. For that matter, I think the test above has the same problem. Both should probably pass in nil for division, or if that doesn't work, the user's division. In fact maybe the filtered_question method should have a default of nil for that argument. A comment making it clear that the division argument represents the division selected in the menu might also be helpful.

Reply:
Yes, you are right, but it is still the correct behavior for the class were it to receive that argument. The class knows nothing about the division drop-down.

The reason I put this in was that I needed some kind of negative test for the addition of the policy. It turns out that checking the loan question policy in the filtered question decorator Does not reduce the available set of questions at all, Assuming the arguments to the decorator are "legal" as you define it. So there is no way to test that The policy is actually being checked without Doing what I did here.

I do think it is still a good idea to check the loan question policy, just in case it changes in the future.